### DISCUSSION TITLE
Automatically convert ISBN-10 to ISBN-13 or ISBN-13 to ISBN-10 on imports.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#155] [Ideas]
I saw in the release note for the December 15, 2021 version. 
- Automatically convert ISBN-10 to ISBN-13 on imports. 

my problem is the opposite, i have ISBN13 but no ISBN10 in my unimarc files

can you make an option in the desired direction
choice 1 - isbn-10 to isbn13
choice 2 - isbn-13 to isbn10

3 files received from the Bibliothèque Départementale inside the duplicates between file 100000 and 110000 (last week's subjects)
with catmandu I transform them into .csv to import them into BiblioteQ via CSV import

[panier.zip](https://github.com/textbrowser/biblioteq/files/7781179/panier.zip)

**Comment by meteos77** [comment]
according to you, is it easy for a c++ novice like me to extract the routine from the file biblioteq_misc_functions.cc
QString biblioteq_misc_functions::isbn10to13(const QString &text)
to make a small bash tool to prepare my import files ?

it will allow me to complete isbn10 and isbn13 whatever the source is.

**Comment by textbrowser** [comment]
Your CSV files require too much thinking and planning. If I perform this automatically (i13 to i10), something else will not work. Review the CVS files and make sure this is what you want.

**Comment by meteos77** [comment]
it's ok i found a ready made program isbn10 to isbn13

my csv file must be formatted well for the csv import to work.
and this operation will have to be done at each new arrival of document (400 -500) every six months.

I have to automate this part as much as possible.

**Comment by textbrowser** [comment]
I'm fine with the conversion. I don't want to revisit this conversion, however, because it creates another problem 10 months later.

**Reply by meteos77** [reply]
precision : this discussion is related to the request of books with multiple volumes.

**Comment by textbrowser** [comment]
The title suggests that it concerns converting i10 to i13.

**Reply by meteos77** [reply]
title Close : Thanks - Automatically convert ISBN-10 to ISBN-13 on imports or t **ISBN-13 to ISBN-10**
